src/core/installAddon.js

```diff
@@ -216,6 +217,7 @@ type WithInstallHelpersProps = {|
   _installTheme: typeof installTheme,
   _tracking: typeof tracking,
   addon: AddonType,
+  currentVersion: AddonVersionType,
```

Comment:
So, I know why you did that but we should not do that. Injecting such props like this is super obscure.

I know we do that for `addon` (which would be tricky to change) and `userAgentInfo` (which could probably be read from Redux as described below) already, but that does not mean we should do that. I was able to remove some props in a previous refactoring, and that was a bit painful, so let's not do that again :p 

I believe we could create a `mapStateToProps` function in the `withInstallHelpers()` HOC: https://github.com/mozilla/addons-frontend/blob/9f51f3ae053720970e7c8d355f00b718b8902045/src/core/installAddon.js#L566-L569

Here is a sketch of patch:

```diff
diff --git a/src/core/installAddon.js b/src/core/installAddon.js
index 0700509df..a255832e7 100644
--- a/src/core/installAddon.js
+++ b/src/core/installAddon.js
@@ -563,8 +563,24 @@ export function withInstallHelpers({
       WrappedComponent,
     )})`;
 
+    const mapStateToProps = (
+      // We cannot use `AppState` because this HOC is shared between `amo` and
+      // `disco`.
+      state: {| versions: VersionsState |},
+      ownProps: WithInstallHelpersProps
+    ) => {
+      const currentVersion = getVersionById({
+        id: ownProps.addon.currentVersionId,
+        state: state.versions,
+      });
+
+      return {
+        currentVersion,
+      };
+    };
+
     return connect(
-      undefined,
+      mapStateToProps,
       _makeMapDispatchToProps({ WrappedComponent, defaultInstallSource }),
     )(WithInstallHelpers);
   };

```

If that works, we could also move `userAgentInfo` there (in a follow-up patch). WDYT?

Reply:
It seems to work well, thanks @willdurand. I've opened https://github.com/mozilla/addons-frontend/issues/6878 as a follow-up.

Reply:
That's great news! 🎉